modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -473,11 +473,19 @@ private List<CompletableFuture<AsyncSqlCursor<List<Object>>>> query0(
             throw new IgniteInternalException(SCHEMA_NOT_FOUND_ERR, format("Schema not found [schemaName={}]", schemaName));
         }
 
-        SqlNodeList nodes = Commons.parse(sql, FRAMEWORK_CONFIG.getParserConfig());
+        CompletableFuture<Void> start = new CompletableFuture<>();
+
+        SqlNodeList nodes = SqlNodeList.EMPTY;
 
         var res = new ArrayList<CompletableFuture<AsyncSqlCursor<List<Object>>>>(nodes.size());
 
-        CompletableFuture<Void> start = new CompletableFuture<>();
+        try {
+            nodes = Commons.parse(sql, FRAMEWORK_CONFIG.getParserConfig());
+        } catch (Throwable th) {
+            start.completeExceptionally(th);
+
+            res.add(CompletableFuture.completedFuture(failedCursor(th)));
```

Comment:
Why don't just return `List.of(CompletableFuture.completedFuture(th))`?
or `List.of(CompletableFuture.completedFuture(failedCursor(th)))` ?

Does it make sense to return failed cursor instead of just failed future?